src/Documents/Session/EntityToJson.ts

```diff
@@ -112,10 +112,29 @@ export class EntityToJson {
                 documentInfo.metadata[CONSTANTS.Documents.Metadata.RAVEN_JS_TYPE] || typeInfo.typeName;
         }
 
+        function differentNestedTypes() {
+            const existing =
+                documentInfo.metadataInstance[CONSTANTS.Documents.Metadata.NESTED_OBJECT_TYPES];
+            if(existing == null)
+                return true;
+            if(Object.keys(existing).length != Object.keys(typeInfo.nestedTypes).length)
+                return true;
+            for (const key in typeInfo.nestedTypes){
+                if(typeInfo.nestedTypes[key] != existing[key]) {
+                    return true;
+                }
+            }
+            return false;
+        }
+
         if (documentInfo.metadataInstance) {
-            documentInfo.metadataInstance[CONSTANTS.Documents.Metadata.NESTED_OBJECT_TYPES] = typeInfo.nestedTypes;
-            documentInfo.metadataInstance[CONSTANTS.Documents.Metadata.RAVEN_JS_TYPE] =
-               documentInfo.metadataInstance["Raven-Node-Type"] || typeInfo.typeName;
+            if(differentNestedTypes()){

```

Comment:
The problem is that here it returns a `Proxy`, so we can't do an easy comparison. I decided to just handle that explicitly, but not sure if this is the best way.